pkg/trace/obfuscate/sql_test.go

```diff
@@ -466,65 +750,162 @@ func TestConsumerError(t *testing.T) {
+		{
+			"USING $A FROM users",
+			`at position 7: prepared statements must start with digits, got "A" (65)`,
+		},
 
-	_, err = obfuscateSQLString("SELECT name, `age}` FROM profile")
-	assert.Error(err)
-	assert.Equal(`at position 17: literal identifiers must end in "`+"`"+`", got "}" (125)`, err.Error())
+		{
+			"USING $09 SELECT",
+			`at position 9: invalid number`,
+		},
 
-	_, err = obfuscateSQLString("SELECT %(asd)| FROM profile")
-	assert.Error(err)
-	assert.Equal(`at position 13: invalid character after variable identifier: "|" (124)`, err.Error())
+		{
+			"INSERT VALUES (1, 2) INTO {ABC",
+			`at position 30: unexpected EOF in escape sequence`,
+		},
 
-	_, err = obfuscateSQLString("USING $A FROM users")
-	assert.Error(err)
-	assert.Equal(`at position 7: prepared statements must start with digits, got "A" (65)`, err.Error())
+		{
+			"SELECT one, :2two FROM profile",
+			`at position 13: bind variables should start with letters, got "2" (50)`,
+		},
 
-	_, err = obfuscateSQLString("USING $09 SELECT")
-	assert.Error(err)
-	assert.Equal(`at position 9: invalid number`, err.Error())
+		{
+			"SELECT age FROM profile WHERE name='John \\",
+			`at position 43: unexpected EOF in string`,
+		},
 
-	_, err = obfuscateSQLString("INSERT VALUES (1, 2) INTO {ABC")
-	assert.Error(err)
-	assert.Equal(`at position 30: unexpected EOF in escape sequence`, err.Error())
+		{
+			"SELECT age FROM profile WHERE name='John",
+			`at position 41: unexpected EOF in string`,
+		},
 
-	_, err = obfuscateSQLString("SELECT one, :2two FROM profile")
-	assert.Error(err)
-	assert.Equal(`at position 13: bind variables should start with letters, got "2" (50)`, err.Error())
+		{
+			"/* abcd",
+			`at position 7: unexpected EOF in comment`,
+		},
 
-	_, err = obfuscateSQLString("SELECT age FROM profile WHERE name='John \\")
-	assert.Error(err)
-	assert.Equal(`at position 42: unexpected EOF after escape character in string`, err.Error())
+		// using mixed cases of backslash escaping the single quote
+		{
+			"SELECT age FROM profile WHERE name='John\\' and place='John\\'s House'",
+			`at position 59: unexpected byte 92`,
+		},
 
-	_, err = obfuscateSQLString("SELECT age FROM profile WHERE name='John")
-	assert.Error(err)
-	assert.Equal(`at position 41: unexpected EOF in string`, err.Error())
+		{
+			"SELECT age FROM profile WHERE place='John\\'s House' and name='John\\'",
+			`at position 69: unexpected EOF in string`,
+		},
+	}
+	for i, tc := range cases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			_, err := NewObfuscator(nil).obfuscateSQLString(tc.query)
+			assert.Error(t, err)
+			assert.Equal(t, tc.expected, err.Error())
+		})
+	}
+}
 
-	_, err = obfuscateSQLString("/* abcd")
-	assert.Error(err)
-	assert.Equal(`at position 7: unexpected EOF in comment`, err.Error())
+func TestLiteralEscapesUpdates(t *testing.T) {
+	cases := []struct {
+		name     string
+		initial  bool
+		query    string
+		success  bool
+		expected bool
```

Comment:
These two are not immediately obvious as to their purpose. Let's either document them by adding comments to their right, or rename them. Maybe `success` can be replaced with `err error` or `err string` (whichever you prefer) and `expected` can be renamed to `final` to pair better with `initial`.

Either renaming, or commenting is fine by me.